### PR TITLE
Rename @ethereumjs/trie to merkle-patricia-tree

### PR DESCRIPTION
**Pull request by holgerd77**
This is a release-preparing PR for the MPT library renaming the package from `@ethereumjs/trie` to `merkle-patricia-tree`.

Unfortunately - as beauty as the new name is - we are running once again into the breaking-release-problem field. A new release within a new naming scope would require breaking releases on all dependants. Beside the changes itself in the MPT library (checkpointing rewrite mainly) are non-breaking and by doing a breaking release here we would exclude all current v4 users from the benefits. So renaming should be postponed to a PiT where a breaking release by feature is at the door.

I would only rename package name references here and keep the folder as `packages/trie` (since - being not in the @ethereumjs naming scope yet, there wouldn't be a 100% appropriate folder naming anyhow).

I am currently not getting the lerna bootstrap process going, after having done a:

```shell
npm run clean
rm package-lock.json
npm install
```

I am getting the following error on bootstrap:

```shell
@ethereumjs/tx: > tsc -p ./tsconfig.browser.json
@ethereumjs/tx: [Browser build] Working... DONE
@ethereumjs/block: > @ethereumjs/block@3.0.0 build /EthereumJS/ethereumjs-monorepo/packages/block
@ethereumjs/block: > ethereumjs-config-ts-build
@ethereumjs/block: [Node build] Using tsconfig.prod.json
@ethereumjs/block: > tsc -p ./tsconfig.prod.json
@ethereumjs/block: [Node build] Working... src/block.ts(3,34): error TS2307: Cannot find module '' or its corresponding type declarations.
@ethereumjs/block: src/block.ts(5,20): error TS2307: Cannot find module '@ethereumjs/common' or its corresponding type declarations.
@ethereumjs/block: src/block.ts(6,40): error TS2307: Cannot find module '@ethereumjs/tx' or its corresponding type declarations.
@ethereumjs/block: src/from-rpc.ts(1,37): error TS2307: Cannot find module '@ethereumjs/tx' or its corresponding type declarations.
@ethereumjs/block: src/header.ts(1,20): error TS2307: Cannot find module '@ethereumjs/common' or its corresponding type declarations.
@ethereumjs/block: src/types.ts(2,20): error TS2307: Cannot find module '@ethereumjs/common' or its corresponding type declarations.
@ethereumjs/block: src/types.ts(3,32): error TS2307: Cannot find module '@ethereumjs/tx' or its corresponding type declarations.
@ethereumjs/block: npm ERR! code ELIFECYCLE
@ethereumjs/block: npm ERR! errno 2
@ethereumjs/block: npm ERR! @ethereumjs/block@3.0.0 build: `ethereumjs-config-ts-build`
@ethereumjs/block: npm ERR! Exit status 2
@ethereumjs/block: npm ERR!
@ethereumjs/block: npm ERR! Failed at the @ethereumjs/block@3.0.0 build script.
@ethereumjs/block: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@ethereumjs/block: npm ERR! A complete log of this run can be found in:
@ethereumjs/block: npm ERR!     /.npm/_logs/2021-02-11T15_16_50_708Z-debug.log
lerna ERR! npm run build exited 2 in '@ethereumjs/block'
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! root@ build: `lerna run build --stream`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the root@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /.npm/_logs/2021-02-11T15_16_50_772Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! root@ bootstrap: `lerna bootstrap --ignore-scripts --include-dependencies --no-ci --hoist && npm run build`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the root@ bootstrap script.
```

@evertonfraga could you have a short look here and see if you spot something? Respectively are there any code or script parts where such a renaming described above might cause problems on the bootstrap process? 🤔

### CODE REVIEW

File: packages/block/src/block.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable no-dupe-class-members */
 
-import { BaseTrie as Trie } from '@ethereumjs/trie'
+import { BaseTrie as Trie } from ''
```
Comment (jochem-brouwer) [inline packages/block/src/block.ts:3]: This won't work 😉

File: packages/client/lib/sync/execution/vmexecution.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Execution, ExecutionOptions } from './execution'
 import { short } from '../../util'
 import VM from '@ethereumjs/vm'
 import { DefaultStateManager } from '@ethereumjs/vm/dist/state'
-import { SecureTrie as Trie } from '@ethereumjs/trie'
+import { SecureTrie as Trie } from ''
```
Comment (jochem-brouwer) [inline packages/client/lib/sync/execution/vmexecution.ts:5]: Same here

File: packages/client/lib/util/debug.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ const level = require('level')
 import Common from '@ethereumjs/common'
 import { Block } from '@ethereumjs/block'
 import VM from './lib'
-import { SecureTrie as Trie } from '@ethereumjs/trie'
+import { SecureTrie as Trie } from ''
```
Comment (jochem-brouwer) [inline packages/client/lib/util/debug.ts:28]: It seems like you changed every @ethereumjs/trie to the empty string...

Comment (holgerd77) [reply packages/client/lib/util/debug.ts:28]: Oh, that might have been close as hard as spotting a consensus error. 😂  😂  😂

head -> table, head -> table, head -> table.

**Comment by holgerd77** [comment]
As a side effect, linting on pushes is now working for me again after all these weeks (months?) 😜

**Comment by holgerd77** [comment]
Ok, this is now ready for review (devp2p CI failure can be ignored, this is just a timing issue independent from the PR which we need to address separately).

**Review by ryanio — APPROVED** [review]
lgtm!

File: packages/trie/package.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@ethereumjs/trie",
+  "name": "merkle-patricia-tree",
   "version": "4.0.0",
   "description": "This is an implementation of the modified merkle patricia tree as specified in the Ethereum's yellow paper.",
```
Comment (ryanio) [inline packages/trie/package.json:4]: just noticed this nit :)
```suggestion
  "description": "This is an implementation of the modified merkle patricia tree as specified in Ethereum's yellow paper.",
```

**Review by ryanio — APPROVED** [review]

**Pull request merged**